bench/app.py

```diff
@@ -140,14 +141,23 @@ def get_app(git_url, branch=None, bench_path='.', skip_assets=False, verbose=Fal
 
 
 def get_app_name(bench_path, repo_name):
-	# retrieves app name from setup.py
-	app_path = os.path.join(bench_path, 'apps', repo_name, 'setup.py')
-	with open(app_path, 'rb') as f:
-		app_name = re.search(r'name\s*=\s*[\'"](.*)[\'"]', f.read().decode('utf-8')).group(1)
-		if repo_name != app_name:
-			apps_path = os.path.join(os.path.abspath(bench_path), 'apps')
-			os.rename(os.path.join(apps_path, repo_name), os.path.join(apps_path, app_name))
-		return app_name
+	app_name = None
+	apps_path = os.path.join(os.path.abspath(bench_path), 'apps')
+	config_path = os.path.join(apps_path, repo_name, 'setup.cfg')
+	if os.path.exists(config_path):
+		config = read_configuration(config_path)
+		app_name = config.get('metadata', {}).get('name')
+
+	if not app_name:
+		# retrieve app name from setup.py as fallback
+		app_path = os.path.join(apps_path, repo_name, 'setup.py')
+		with open(app_path, 'rb') as f:
+			app_name = re.search(r'name\s*=\s*[\'"](.*)[\'"]', f.read().decode('utf-8')).group(1)
+
+	if app_name and repo_name != app_name:
+		os.rename(os.path.join(apps_path, repo_name), os.path.join(apps_path, app_name))
+
+	return app_name
```

Comment:
if `app_name` is None,  `get_app_name` would return None too; this breaks 

https://github.com/frappe/bench/blob/52ee67f298adb9939d9cee1266250968f35e8c0b/bench/app.py#L145-L146

```suggestion
		return app_name

	return repo_name
```